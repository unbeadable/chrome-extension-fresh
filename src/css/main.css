.unbdbl-badge {
    display: flex;
    border: 1px solid #bc2525;
    background-color: #bc2525;
    border-radius: 5px;
    padding: 4px;
    width: 100%;
    max-width: 250px;
    box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 15px;
    position: relative;
    margin-top: 15px;
}

.unbdbl-primary-text {
    flex: 1;
    color: white;
    font-size: 13px;
    line-height: normal;
    padding-left: 5px;
}

.unbdbl-badge-icon {
    flex: 0 0 24px;
}