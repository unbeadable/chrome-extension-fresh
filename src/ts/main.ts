export function render() {
    const findAsin = () => {
        let elementsByClassName = document.getElementsByClassName('label');
        const asinLabel = Array.from(elementsByClassName)
            .find(it => it.innerHTML == "ASIN");

        if (!!asinLabel) {
            const nextElementSibling = asinLabel.nextElementSibling;
            if (!!nextElementSibling) {
                return nextElementSibling.innerHTML
            }
        }

        return null;
    };

    const badgeToInject = document.createElement('div');
    badgeToInject.setAttribute('id', 'badge');
    const hostDiv = document.getElementById('zeitgeistBadge_feature_div');

    if(!!findAsin()) {
        hostDiv.appendChild(badgeToInject);
    }
}

render();

