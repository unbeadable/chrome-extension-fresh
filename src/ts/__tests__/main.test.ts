import { render } from '../main';

describe('main', () => {
    const hostDivId = 'zeitgeistBadge_feature_div';
    const hostDivHtml: String = `<div id="${hostDivId}"></div>`;
    const asinInfo =
        '<table><tr><td class="label">ASIN</td><td class="value">B001LQCCDK</td></tr></table>';

    it('should attach badge to target host div when page has ASIN info', async () => {
        document.body.innerHTML = hostDivHtml + asinInfo;

        render();

        const hostDiv: HTMLElement = document.getElementById(hostDivId);
        expect(hostDiv.innerHTML).toMatchSnapshot();
    });

    it('should keep HTML as is when page does not have ASIN info', async () => {
        document.body.innerHTML = hostDivHtml;

        render();

        const hostDiv: HTMLElement = document.getElementById(hostDivId);
        expect(hostDiv.innerHTML).toMatchSnapshot();
    });
});
