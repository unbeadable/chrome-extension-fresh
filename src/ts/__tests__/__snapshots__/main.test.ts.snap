// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main should attach badge to target host div when page has ASIN info 1`] = `"<div class=\\"unbdbl-badge unbdbl-primary-text\\"><img src=\\"https://s3.eu-central-1.amazonaws.com/unbeadable/baseline-report-24px.svg\\" class=\\"unbdbl-badge-icon\\"><span class=\\"unbdbl-primary-text\\"></span></div>"`;

exports[`main should keep HTML as is when page does not have ASIN info 1`] = `""`;
